src/main/java/com/google/songgame/data/YoutubeParser.java

```diff
@@ -0,0 +1,161 @@
+  private static final String APPLICATION_NAME = "Song Guessing Game";
+  private static final JsonFactory GSON_FACTORY = GsonFactory.getDefaultInstance();
+  private static final long MAX_RESULTS = 25L;
+
+  public ArrayList<String> getPlaylistVideoIds(String playlistUrl) {
+    String playlistId = getPlaylistIdFromUrl(playlistUrl);
+    PlaylistItemListResponse playlistItem = new PlaylistItemListResponse();
+    // Retrieve Playlist item from Youtube API
+    try {
+      playlistItem = getPlaylistInfo(playlistId);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    // Parse Playlist item Json string to retrieve video IDs
+    String playlistItemJson = new Gson().toJson(playlistItem);
+    ArrayList<String> playlistVideoIds = parseVideoIdsFromPlaylistItem(playlistItemJson);
+    return playlistVideoIds;
+  }
+
+  /**
+   * Returns playlist ID
+   */
+  private String getPlaylistIdFromUrl(String playlistUrl) {
+    if (playlistUrl.contains("youtube.com/playlist?list=")) {
+      int start = playlistUrl.indexOf("list=") + 5;
+      int end = playlistUrl.length();
+      if (playlistUrl.contains("&")) {
+        end = playlistUrl.indexOf("&", start);
+      }
+      return playlistUrl.substring(start, end);
+    } else {
+      System.err.println(playlistUrl + " is not a valid YouTube Playlist URL.");
+      return "";
+      // TODO @hdee: do something more complicated to handle this error.
+      // TODO @hdee: test this method.
+    }
+  }
+
+   /**
+   * Call function to create API service object.
+   *
+   * @throws GeneralSecurityException, IOException, GoogleJsonResponseException
+   */
+  private static PlaylistItemListResponse getPlaylistInfo(String playlistId)
+      throws GeneralSecurityException, IOException, GoogleJsonResponseException {
+    YouTube youtubeService = getService();
+    // Define and execute the API request
+    YouTube.PlaylistItems.List request =
+        youtubeService
+            .playlistItems()
+            .list("snippet")
+            .setMaxResults(MAX_RESULTS)
+            .setPlaylistId(playlistId);
+    PlaylistItemListResponse response = request.execute();
+    return response;
+  }
+
+  /**
+   * Returns an ArrayList of video IDs
+   * TODO @hdee: add tests for this method
+   */
+  private ArrayList<String> parseVideoIdsFromPlaylistItem(String playlistItemJson) {
+    String[] playlistItemData = playlistItemJson.split("\",\"");
+    ArrayList<String> playlistVideos = new ArrayList<String>();
+    // extract video ID from sections
+    for (String data : playlistItemData)
+      if (data.startsWith("videoId\":\"")) {
+        int idStart = data.indexOf("\":\"") + 3;
+        int idEnd = data.indexOf("\"", idStart);
+        String videoId = data.substring(idStart, idEnd);
+        playlistVideos.add(videoId);
+      }
+    return playlistVideos;
+  }
+
+  public Video getRandomVideoFromPlaylist(ArrayList<String> playlistVideoIds) {
```

Comment:
Could you add a javadoc to this method as well? It looks like many of the private methods have javadocs, which is fine, but really javadocs are most important on public methods, since those are meant to be used by other classes which use this class.